helix-core/src/main/java/org/apache/helix/BucketDataAccessor.java

```diff
@@ -0,0 +1,35 @@
+package org.apache.helix;
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.IOException;
+
+
+public interface BucketDataAccessor {
+
+  <T extends HelixProperty> boolean bucketWrite(String path, T value);
+
+  HelixProperty bucketRead(String path);
+
+  <T extends HelixProperty> boolean compressedBucketWrite(String path, T value) throws IOException;
+
+  HelixProperty compressedBucketRead(String path);
+
+  void setBucketSize(int size);
```

Comment:
Shall we just final this var in the constructor?
Allow updating might increase the complicity.

Reply:
If it's an abstract class, the field can be of a field. Not a good idea for external to update at will

Reply:
I agree. To make this simple, I will let the user define it in the constructor.